### PR TITLE
(maint) Merge 5.5.x -> 6.0.x

### PR DESCRIPTION
**Pull request by mihaibuzgau**

### CODE REVIEW

File: acceptance/lib/puppet/acceptance/common_utils.rb [hunk 1]
```diff
@@ -30,6 +30,16 @@ def user_absent(host, username)
       end
     end
 
+    module PackageUtils
```
Comment (gimmyxd) [inline acceptance/lib/puppet/acceptance/common_utils.rb]: - did not promote to 60.x - CAUtils, CronUtils
    - added PackageUtils @loopinu

File: lib/puppet/provider/package/pip.rb [hunk 2]
```diff
@@ -2,162 +2,123 @@
 # <http://pip.pypa.io/>
 
 require 'puppet/provider/package'
+require 'puppet/provider/package_targetable'
```
Comment (gimmyxd) [inline lib/puppet/provider/package/pip.rb]: @tkishel can you double check that we got the correct funtionality

**Comment by DavidS** [comment]
The fixes from `device.rb` are also needed in the 6.0.x branch. I've just confirmed the faulty behaviour with 6.0.10 from puppet-enterprise-2019.0.4-rc1-47-gb975b36-ubuntu-18.04-amd64 . I can provide an updated device.rb later today.

**Comment by gimmyxd** [comment]
Jenkins please test this

**Comment by jhelwig** [comment]
The acceptance failure is in one of the device tests:

```
08:12:36 Failed Tests Cases:
08:12:36   Test Case tests/puppet_device_test.rb reported: #<Minitest::Assertion: cert requesting failed.
08:12:36 </Permission denied/> expected to not match
08:12:36 <"\e[1;31mError: Could not request certificate: Could not write /opt/puppetlabs/puppet/cache/devices/spinny1.example.com/ssl/private_keys/spinny1.example.com.pem to privatekeydir: Permission denied @ rb_sysopen - /opt/puppetlabs/puppet/cache/devices/spinny1.example.com/ssl/private_keys/spinny1.example.com.pem\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/file_system/file_impl.rb:31:in `initialize'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/file_system/file_impl.rb:31:in `open'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/file_system/file_impl.rb:31:in `open'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/file_system.rb:47:in `open'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/settings/directory_setting.rb:15:in `block in open_file'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/settings/file_setting.rb:241:in `block (2 levels) in controlled_access'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util.rb:142:in `withumask'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/settings/file_setting.rb:231:in `block in controlled_access'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/suidmanager.rb:74:in `asuser'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/settings/file_setting.rb:229:in `controlled_access'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/settings/directory_setting.rb:14:in `open_file'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/indirector/ssl_file.rb:147:in `write'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/indirector/ssl_file.rb:78:in `save'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/indirector/key/file.rb:34:in `save'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/indirector/indirection.rb:285:in `save'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/ssl/host.rb:80:in `generate_key'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/ssl/host.rb:121:in `certificate'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/ssl/host.rb:305:in `wait_for_cert'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/device.rb:390:in `setup_host'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/device.rb:279:in `block (3 levels) in main'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/context.rb:65:in `override'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet.rb:252:in `override'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/device.rb:258:in `block (2 levels) in main'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/device.rb:256:in `each'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/device.rb:256:in `collect'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/device.rb:256:in `block in main'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/context.rb:65:in `override'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet.rb:252:in `override'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/device.rb:241:in `main'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:390:in `run_command'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:382:in `block in run'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util.rb:671:in `exit_on_fail'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:382:in `run'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/command_line.rb:139:in `run'\n/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/command_line.rb:77:in `execute'\n/opt/puppetlabs/puppet/bin/puppet:5:in `<main>'\e[0m\n">.>
08:12:36     Test line: tests/puppet_device_test.rb:48:in `block (2 levels) in run_test'
```

File: lib/puppet/provider/package/pip.rb (hunk 2 above)
Comment (tkishel) [reply lib/puppet/provider/package/pip.rb]: Looks good! 6.0.x introduces `PUP-5985 Don't use features for windows?` and this mergup keeps that and maintains the targetable functionality. I double-checked this file ... and all the others in `(PUP-1082) targetable package command`.

**Comment by gimmyxd** [comment]
Jenkins please test this on osx1014-64a,redhat7-64a,ubuntu1604-64a,windows2012r2-64a

**Comment by gimmyxd** [comment]
Jenkins please test this on osx1014-64a,redhat7-64a,ubuntu1604-64a,windows2012r2-64a

**Comment by DavidS** [comment]
I need to prep for Contributor Summit next week and won't be available for this work until Jun 10th. cc @davidmalloncares

**Comment by gimmyxd** [comment]
@joshcooper can we fix that in a following PR since it does not seem related to the merge-up? In this way we will be able to unblock the pipeline.

From what i see `pool = Puppet::Network::HTTP::Pool.new(Puppet[:http_keepalive_timeout])` is already there, since those lines were not modified.

**Comment by joshcooper** [comment]
I'll take a look at this today

**Pull request merged**